Stamina

Rohesie opened the pull request:
* Damage-based movespeed uses the new system.
* A ton of duplicate `updatehealth()` removed.
* Some damage procs cleaned up.
* Humans handle damage through the species datum.
* handle_suit_punctures() and other ancient life stuff, or unused procs/defines, gone.

## Changelog
:cl:
add: Stamina type of damage added. It comes and goes faster than other damage types, knocksdown the victim if it has too much of it, and is otherwise harmless.
tweak: Walk mode made severely faster, and the default for humans. They can run by making use of a stamina buffer, which they recover when not running. Synths don't have stamina, so they cannot sprint/run but are unaffected by stamina damage.
balance: Xenos now have melee armor. Humans also have melee accuracy, which is perfect while walking and only 80% when running. Their tackles now  fullyrespect melee armor.
balance: Neurotoxin no longer injects chems. It now deals direct stamina-damage, same as xeno tackles, resin spit and a few other things.
add: Examine is now a keybind, defaulted to Shift. It merges old shift+click and alt+click on a turf near you. Alt defaults to run-while-holding, and you can shoot or attack while doing so. You can rebind it as well.
balance: The hunter is now much faster while stealthed, so as a consequence its sneak attack has been nerfed some. Probably needs further tweaks.
/:cl:

spookydonut commented on code/__DEFINES/conflict.dm:
```diff
@@ -38,8 +38,8 @@
-#define AMMO_XENO_ACID 			(1<<1)
-#define AMMO_XENO_TOX			(1<<2)
+#define AMMO_XENO 				(1<<1)
+//#define AMMO_XENO_TOX			(1<<2) //Unused value.
```
delete

spookydonut replied:
fair

spookydonut replied:
just uncomment it and leave the unused comment

Rohesie commented on code/modules/mob/living/carbon/xenomorph/castes/crusher/abilities_crusher.dm:
```diff
@@ -133,10 +133,10 @@
 		var/armor_block = L.run_armor_check("chest", "melee")
 		if(ishuman(L))
 			var/mob/living/carbon/human/H = L
```
Legacy of old code. Fixed.

spookydonut commented:
@Rohesie merge conflict

Rohesie commented:
This one depends on #3119

Rohesie commented:
This needs a lot of balancing, but the base is functional.

Eskjjlj commented:
After runners and hunters pounce their move intent is automatically switched to Stalk which is the xeno walk mode.

Rohesie commented:
> After runners and hunters pounce their move intent is automatically switched to Stalk which is the xeno walk mode.

Fixed.

GravitronBlue commented:
Assuming this PR is responsible I'm not a fan of how the shift-examine and alt key turf menu are combined into a single hotkey.

GravitronBlue commented on code/datums/keybinding/mob.dm:
```diff
@@ -146,18 +146,64 @@
+	var/turf/examined_turf = get_turf(A)
+	if(examined_turf && user.TurfAdjacent(examined_turf))
+		user.listed_turf = examined_turf
+		user.client.statpanel = examined_turf.name
```
Please consider separating this into its own client preference hotkey or just being used by alt-clicking like it was before.

Rohesie replied:
Can you expand on the reasons a bit?

GravitronBlue replied:
From a marine standpoint the new default alt key being used to examine messes with certain inventory shortcuts like alt-clicking an MRE or pill bottle to grab the first item out from it.
Though this can be avoided by changing keybindings it ruins consistency between here and TG by having to use a different key just to pull up the turf contents menu.

The turf menu also pops up when you examine inventory items, which is a bit of a nuisance when you just want to check gun stats.
There's probably some actual cases where this could be disruptive, like trying to grab a medical scan from a patient while sorting through a pile of medkits on a nearby table.

I've also been unable to examine things using shift-click while as a ghost.

Personally just think this is a specific change that wasn't asked for, though perhaps you'll find a different opinion and support on the server.

davidchan13 commented:
Synths still can't run.

GravitronBlue commented:
The way stamina damage is applied to barricades and structures needs to be examined, as sentinels are dealing massage damage to fortifications.

Isy232 commented:
Xenos are by default on "stalk" mode. This gets reset every time they evolve. Most xenos are not used to this and I myself can blame several deaths on this, please fix this.

Rohesie commented:
> The way stamina damage is applied to barricades and structures needs to be examined, as sentinels are dealing massage damage to fortifications using neurospit.

Fixed, objects no longer take bio damage, they have 100 bio armor.

> Xenos are by default on "stalk" mode. This gets reset every time they evolve. Most xenos are not used to this and I myself can blame several deaths on this, please fix this.
> 
> Please just remove stalk from all xenos that don't use it, it exists just to be accidentally pressed and screw you over as it is now.

Fixed. They weren't on stalk mode but changed to it on the toggle intent hotkey, which now toggles it to stalk and goes back to running when released.
